pkg/operator/ceph/provisioner/controller/controller.go

```diff
@@ -1082,25 +1089,25 @@ func (ctrl *ProvisionController) scheduleOperation(operationName string, operati
 	}
 }
 
-func (ctrl *ProvisionController) getStorageClassFields(name string) (string, map[string]string, error) {
+func (ctrl *ProvisionController) getStorageClassFields(name string) (string, *v1.PersistentVolumeReclaimPolicy, map[string]string, error) {
 	classObj, found, err := ctrl.classes.GetByKey(name)
 	if err != nil {
-		return "", nil, err
+		return "", nil, nil, err
 	}
 	if !found {
-		return "", nil, fmt.Errorf("StorageClass %q not found", name)
+		return "", nil, nil, fmt.Errorf("StorageClass %q not found", name)
 		// 3. It tries to find a StorageClass instance referenced by annotation
 		//    `claim.Annotations["volume.beta.kubernetes.io/storage-class"]`. If not
 		//    found, it SHOULD report an error (by sending an event to the claim) and it
 		//    SHOULD retry periodically with step i.
 	}
 	switch class := classObj.(type) {
 	case *storage.StorageClass:
-		return class.Provisioner, class.Parameters, nil
+		return class.Provisioner, class.ReclaimPolicy, class.Parameters, nil
```

Comment:
Could you replace this line with `fallthrough` as it is the same as 1108.

Reply:
`cannot fallthrough in type switch`

Reply:
My bad doesn't work in that case.